vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt

```diff
@@ -175,78 +182,79 @@ class TimelineViewModel @AssistedInject constructor(
     }
 
     init {
-        timeline.start(initialState.rootThreadEventId)
-        timeline.addListener(this)
-        observeRoomSummary()
-        observeMembershipChanges()
-        observeSummaryState()
-        getUnreadState()
-        observeSyncState()
-        observeDataStore()
-        observeEventDisplayedActions()
-        observeUnreadState()
-        observeMyRoomMember()
-        observeActiveRoomWidgets()
-        observePowerLevel()
-        setupPreviewUrlObservers()
-        room.getRoomSummaryLive()
-        viewModelScope.launch(Dispatchers.IO) {
-            tryOrNull { room.readService().markAsRead(ReadService.MarkAsReadParams.READ_RECEIPT) }
-        }
-        // Inform the SDK that the room is displayed
-        viewModelScope.launch(Dispatchers.IO) {
-            tryOrNull { session.roomService().onRoomDisplayed(initialState.roomId) }
-        }
-        callManager.addProtocolsCheckerListener(this)
-        callManager.checkForProtocolsSupportIfNeeded()
-        chatEffectManager.delegate = this
-
-        // Ensure to share the outbound session keys with all members
-        if (room.roomCryptoService().isEncrypted()) {
-            rawService.withElementWellKnown(viewModelScope, session.sessionParams) {
-                val strategy = it.getOutboundSessionKeySharingStrategyOrDefault()
-                if (strategy == OutboundSessionKeySharingStrategy.WhenEnteringRoom) {
-                    prepareForEncryption()
+        viewModelScope.launch {
+            timeline.await().also {
+                it.start(initialState.rootThreadEventId)
+                it.addListener(this@TimelineViewModel)
```

Comment:
is it possible for start to trigger the listener before it's finished being added?